src/api/verification_case.py

```diff
@@ -0,0 +1,167 @@
+            loaded_cases = json.load(f)
+
+        # check if the input dict is in the right format
+        # will employ 'validate_verification_case_structure' static method once it's added
+
+        for ele in loaded_cases["cases"]:
+            case_suite[uuid.uuid1()] = ele
+
+        return case_suite
+
+    def load_verification_cases_from_json(self, json_case_path: str) -> list:
+        """Add verification cases from specified json file into self.case_suite
+
+        Args:
+            json_case_path: str, path to the json file containing fully defined verification cases.
+
+        Returns:
+          List, unique ids of verification cases loaded in self.case_suite
+        """
+
+        # check if json_case_path type is str
+        if isinstance(json_case_path, str):
+            # check if the json_case_path exists
+            if os.path.isfile(json_case_path):
+                self.case_suite = self.read_case(json_case_path, self.case_suite)
+            else:
+                logging.error(
+                    "Please make sure that the 'json_case_path' argument is correct."
+                )
+        else:
+            logging.error(
+                f"The type of the 'json_case_path' argument has to be str, but {type(json_case_path)} type is provided. Please verify the 'json_case_path' argument."
+            )
+
+        return list(self.case_suite.keys())
+
+    def save_case_suite_to_json(self, json_path: str, case_ids: list = []):
+        """Save verification cases to a dedicated file. If case_ids is empty, all the cases in `self.case_suite` is saved. If case_ids includes specific cases' hash, the hashes in the list are only saved.
+
+        Args:
+            json_path: str. path to the json file to save the cases.
+            case_ids: List. Unique ids of verification cases to save. By default, save all cases in `self.case_suite`
+        """
+
+        # check json_path type
+        if not isinstance(json_path, str):
+            logging.error(
+                f"The type of the 'json_path' argument has to be str, but {type(json_path)} type is provided. Please verify the 'json_path' argument."
+            )
+            return None
+
+        # check case_ids type
+        if not isinstance(case_ids, list):
+            logging.error(
+                f"The type of the 'case_ids' argument has to be str, but {type(case_ids)} type is provided. Please verify the 'case_ids' argument."
+            )
+            return None
+
+        # if json_path is saved in a different dir, check if it exists
+        json_path_split = json_path.split("/")
+        dir_path = "/".join(json_path_split[:-1])
+        if len(dir_path) > 2:  # when json_path includes directory path
+            if not os.path.exists(dir_path):
+                logging.error(
+                    f"{dir_path} directory doesn't exist. Please make sure to provide an already existing directory."
+                )
+                return None
```

Comment:
1. Added the JSON file path-checking part.
2. When ```len(dir_path) <= 2```, it means the ```json_case_path``` doesn't include dir (only includes file name e.g., testing.json).
Example: 
i) if json_path = "testing.json",  dir_path = ".", len(dir_path ) = 1
ii) if json_path = "./testing.json",  dir_path = "", len(dir_path ) = 0
iii) if json_path3 = "./testing1/testing.json", dir_path = "./testing1", len(dir_path ) = **10**

This part is further updated as shown below.
```python
dir_path = json_path.split("/")[-1:]
```
This makes the above part to
i) if json_path = "testing.json",  dir_path = ".", len(dir_path ) = 1
ii) if json_path = "./testing.json",  dir_path = "", len(dir_path ) = 0
iii) if json_path3 = "./testing1/testing.json",  dir_path = "./testing1", len(dir_path ) = **2**


3. I believe the case when ```/```s aren't used is also considered.

Question) If we decided to let Python handle the file (path) existence, can we not include all the ```os.isfile``` and ```os.path.exists``` parts in our API design?

Reply:
to point 2. what about "." or "./" or "/"?

Reply:
Great point.

Based on your discussion last time, I realized that this approach would cause some issues if users use different os (windows, mac, linux) because as far as know each os deals with paths a bit differently (e.g., /, //, \\, etc.). Hence, I'll leave out the logic about checking whether a directory exists or not. and let Python deal with this.

Please let me know if you have a different thought.

Reply:
Great point.

Based on your discussion last time, I realized that this approach would cause some issues if users use different os (windows, mac, Linux) because as far as know each os deals with paths a bit differently (e.g., /, //, \, etc.). Hence, I'll leave out the logic about checking whether a directory exists or not. and let Python deal with this.

Please let me know if you have a different thought.